include/envoy/http/codec.h

```diff
@@ -52,6 +53,12 @@ class StreamEncoder {
    * @return Stream& the backing stream.
    */
   virtual Stream& getStream() PURE;
+
+  /**
+   * Encode METADATA.
+   * @param metadata_map is the METADATA to encode.
+   */
+  virtual void encodeMetadata(const MetadataMap& metadata_map) PURE;
```

Comment:
I think this is a good PR to start landing docs with, even if we have them tagged as work in progress.
Let's talk about when you can encodeMetadata, how often you can encode it, when it will be passed up to filters (we can edit the docs if they change when that lands) etc.

Reply:
Still working on the doc.